src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -1959,16 +1959,27 @@ TEST_CASE("settings upgrade", "[tx][soroban][upgrades]")
         auto const& contractID = contractKeys[0].contractData().contract;
 
         // build upgrade
-        auto costKey = configSettingKey(
-            ConfigSettingID::CONFIG_SETTING_CONTRACT_LEDGER_COST_V0);
 
-        ConfigSettingEntry cost{};
+        // This test assumes that all settings including and after
+        // CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW are not upgradeable, so they
+        // won't be included in the upgrade.
+        xdr::xvector<ConfigSettingEntry> updatedEntries;
+        for (uint32_t i = 0;
+             i < static_cast<uint32_t>(CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW);
+             ++i)
         {
             LedgerTxn ltx(app->getLedgerTxnRoot());
-            auto costEntry = ltx.load(costKey);
-            cost = costEntry.current().data.configSetting();
+            auto costEntry =
+                ltx.load(configSettingKey(static_cast<ConfigSettingID>(i)));
+            updatedEntries.emplace_back(
+                costEntry.current().data.configSetting());
         }
 
+        // Update one of the settings. The rest will be the same so will not get
+        // upgraded, but this will still test that the limits work when writing
+        // all settings to the contract.
+        auto& cost = updatedEntries.at(static_cast<uint32_t>(
+            ConfigSettingID::CONFIG_SETTING_CONTRACT_LEDGER_COST_V0));
```

Comment:
It's unfortunate that we're using ConfigSettings for non-config values... we probably don't have time for this, but it'd be nice to clean this up.

Reply:
Yeah we do